react/src/components/Routes.js

```diff
@@ -10,8 +10,12 @@ import ADP from './ADP';
 import Login from "./Login";
 import CreateAccount from "./CreateAccount";
 import NotFound from './NotFound';
+import UserProfile from "./UserProfile";
+
+const useQuery = () => new URLSearchParams(useLocation().search);
```

Comment:
This just adds handling for ?foo=xyz&bar=abc type of links, right? I was wondering how that was done.

Reply:
Yep. So if/when we want to add more, we need to handle it using `useQuery` like we do in the `/profile` page.